ksql-rest-app/src/test/java/io/confluent/ksql/rest/util/ClusterTerminatorTest.java

```diff
@@ -0,0 +1,245 @@
+    // Given:
+    givenSinkTopicExists("FOO", "K_FOO");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Foo"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(Collections.emptyList());
+  }
+
+
+  @Test
+  public void shouldDeleteTopicListWithPattern() {
+    //Given:
+    givenSinkTopicsExist("K_FO", "K_FOO", "K_FOOO", "NotMatched");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_FO.*"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(ImmutableList.of("K_FOOO", "K_FO", "K_FOO"));
+  }
+
+  @Test
+  public void shouldThrowIfCouldNotDeleteTopicListWithPattern() {
+    // Given:
+    givenSinkTopicExists("FOO", "K_FOO");
+    doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .when(kafkaTopicClient).deleteTopics(Collections.singletonList("K_FOO"));
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("Exception while deleting topics: K_FOO");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_FO*"));
+
+  }
+
+  @Test
+  public void shouldThrowIfDeleteTopicListMatchesNonSink() {
+    // Given:
+    givenNonSinkTopicExits("BAR", "BAR");
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("Invalid request: BAR is not a KSQL sink topic.");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("BAR"));
+  }
+
+  @Test
+  public void shouldDeleteCommandTopic() {
+    // When:
+    clusterTerminator.terminateCluster(Collections.emptyList());
+
+    // Then:
+    final InOrder inOrder = Mockito.inOrder(kafkaTopicClient, ksqlEngine);
+    inOrder.verify(ksqlEngine).getPersistentQueries();
+    inOrder.verify(kafkaTopicClient)
+        .deleteTopics(Collections.singletonList("_confluent-ksql-command_topic_command_topic"));
+    inOrder.verify(ksqlEngine).close();
+  }
+
+  @Test
+  public void shouldThrowIfCannotDeleteCommandTopic() {
+    // Given:
+
+    doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .when(kafkaTopicClient)
+        .deleteTopics(Collections.singletonList("_confluent-ksql-command_topic_command_topic"));
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage(
+        "Could not delete the command topic: _confluent-ksql-command_topic_command_topic");
+
+    // When:
+    clusterTerminator.terminateCluster(Collections.emptyList());
+
+    // Then:
+    verify(kafkaTopicClient)
+        .deleteTopics(Collections.singletonList("_confluent-ksql-command_topic_command_topic"));
+  }
+
+  private static KsqlTopic getKsqlTopic(final String topicName, final String kafkaTopicName,
```

Comment:
`topicName` is not important to these tests. It could be anything and the tests would still pass. So remove it.

Just pass a constant to the constuctor `"registered-topic-name"` or if you want `"registered" + kafkaTopicName`.